### ISSUE TITLE
Doubleclick to edit custom property key

### ISSUE THREAD

**Issue by issy123** [bjorn/tiled#1662]
### Request
It would be nice if you simply could double click on a custom property key to edit the key. Just like when you doubleclick a custom text property to edit it's value.

### Reason

Convenience!

**Comment by bjorn** [comment]
Yep, I agree. I think having it editable inline like the property value is not going to be easy, but alternatively maybe we could popup the rename dialog when the name is double-clicked.

**Comment by issy123** [comment]
You could add three dots next to it(like editing text single line vs multi line text), so if you want to change the datatype you can click that to have a pop up. So you could still have the convenience of inline editing.

**Comment by bjorn** [comment]
The data type can be changed by right-clicking the property. The problem with inline editing of the name is that the Qt Property Browser solution, which is has quite a large and complicated code base, does not support this feature. But that's where it would need to be added.

**Comment by bjorn** [comment]
This might be easier to implement now with the new properties framework added in #4045.

**Comment by bkvarma07** [comment]
Hi! I would like to work on this issue. I have already built Tiled locally and would love to contribute. Could you guide me on where to start?

**Comment by bjorn** [comment]
@bkvarma07 You're welcome to have a look into this! Are you already familiar with Tiled? The Properties view is a dock widget containing a `PropertiesWidget` instance. The property names are displayed using `PropertyLabel` widgets, and I think you might be able to handle the double-click there by overriding `mouseDoubleClickEvent(QMouseEvent *event)`.

An initial implementation could just trigger a `renameRequested` signal from the `Property` that opens the rename dialog, but ideally the property name would be editable in-place (similar to the line edit created by the `AddValueProperty::createLabel`).

~~A tricky thing might be to make sure this only works on the empty area below the last custom property, because I don't think we want this behavior on scrollbars or other widgets in the view.~~ (sorry, confused this with #845)

**Comment by anandkr30-glitch** [comment]
Hello!
I want to work on this(#1662) for GSoC 2026.
So basically I need to let users double-click the property KEY 
to edit it, just like they can with the VALUE, right?
I saw there's already code for double-clicking VALUES. 
I'm thinking I can find that code and do something similar for KEYS.
Is that the right approach? Any guidance on which file to look at? and also how should i proceed with the things if there is a better way exisiting it?
Thanks!

**Comment by bjorn** [comment]
@anandkr30-glitch I just tried to give some guidance to @bkvarma07 on this. While you're welcome to have a look into this as well, it might be better to look for another issue to work on to avoid duplicate efforts.

**Comment by bkvarma07** [comment]
Hi @bjorn, thanks for the guidance!
I’ve implemented double-click detection on `PropertyLabel` and I’m now working on triggering rename functionality for the property key.

I’ll share a PR soon once I have a working version. Let me know if you have any suggestions!

**Comment by bkvarma07** [comment]
Hi! I've submitted a PR implementing a dialog-based rename on double-click. The CI checks are currently running. Happy to improve it further (for example, inline editing) if needed.
